lib/internal/process/promises.js

```diff
@@ -60,11 +58,23 @@ function setupPromises(scheduleMicrotasks) {
   }
 
   function emitWarning(uid, reason) {
+    try {
+      if (reason instanceof Error) {
+        process.emitWarning(reason.stack, 'UnhandledPromiseRejectionWarning');
```

Comment:
So I guess the only downfall to doing it this way is it won't include any additional properties added on to the Error object. Any reason to do this vs using `util.inspect()`?

Reply:
Just that I wasn't aware of `util.inspect()`. I went with what the rest of the code did in the same function.

Reply:
Namely, Node doesn't show this information for synchronous errors:

```js
// test.js
var e = new Error("error");
e.foo = "bar";
throw e;
```
```bash
~/test/test.js:3
throw e;
^

Error: error
    at Object.<anonymous> (/Users/benjamin/Downloads/test.js:1:71)
    at Module._compile (module.js:573:30)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)
    at Function.Module._load (module.js:462:3)
    at Function.Module.runMain (module.js:609:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:598:3
```